File: generated_test_scripts/testcases/test_case_2.py

from selenium import webdriver
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time

class LoginPage:
    def __init__(self, driver):
        self.driver = driver

    def login(self, username, password):
        username_field = WebDriverWait(self.driver, 10).until(
            EC.visibility_of_element_located((By.XPATH, '//*[@id="user-name"]')))
        password_field = self.driver.find_element(By.XPATH, '//*[@id="password"]')
        login_button = self.driver.find_element(By.XPATH, '//*[@id="login-button"]')

        username_field.send_keys(username)
        time.sleep(3)
        password_field.send_keys(password)
        time.sleep(3)
        login_button.click()
        time.sleep(3)

def main():
    options = Options()
    options.add_argument('--headless')
    options.add_argument('--disable-notifications')
    options.add_argument('--incognito')
    options.add_argument('--window-size=1920,1080')

    try:
        driver = webdriver.Chrome(options=options)

        driver.get("https://saucedemo.com/")
        time.sleep(5)
        
        login_page = LoginPage(driver)
        login_page.login("standard_user", "secret_sauce")

        bike_light_button = WebDriverWait(driver, 10).until(
            EC.visibility_of_element_located((By.XPATH, '//*[@id="add-to-cart-sauce-labs-bike-light"]')))
        fleece_jacket_button = driver.find_element(By.XPATH, '//*[@id="add-to-cart-sauce-labs-bolt-t-shirt"]')

        bike_light_button.click()
        time.sleep(3)
        fleece_jacket_button.click()
        time.sleep(3)

        cart_count = WebDriverWait(driver, 10).until(
            EC.visibility_of_element_located((By.XPATH, '//*[@id="shopping_cart_container"]/a/span')))
 
        assert cart_count.text == '2'
        driver.quit()
        exit(0)

    except Exception as e:
        print(f"Test failed: {e}")
        driver.quit()
        exit(1)

if __name__ == "__main__":
    main()

File: generated_test_scripts/poc_data - sheet1/test_case_2.java

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.util.concurrent.TimeUnit;

public class UITest {
    public static void main(String[] args) {
        System.setProperty("webdriver.chrome.driver", "path/to/chromedriver");

        ChromeOptions options = new ChromeOptions();
        options.addArguments("--headless");
        options.addArguments("--disable-notifications");
        options.addArguments("--disable-popup-blocking");
        options.addArguments("--incognito");

        WebDriver driver = new ChromeDriver(options);
        WebDriverWait wait = new WebDriverWait(driver, 10);

        try {
            driver.get("http://yourpageurl.com");
            driver.manage().window().maximize();
            TimeUnit.SECONDS.sleep(5);

            By usernameLocator = By.xpath("//input[@name='username']");
            By passwordLocator = By.xpath("//input[@name='password']");
            By submitButtonLocator = By.xpath("//button[@type='submit']");
            By errorMessageLocator = By.xpath("//*[contains(text(), 'invalid username')]");

            // Wait for elements to appear and interact
            wait.until(ExpectedConditions.visibilityOfElementLocated(usernameLocator));
            driver.findElement(usernameLocator).sendKeys("incorrectUser");
            TimeUnit.SECONDS.sleep(3);

            wait.until(ExpectedConditions.visibilityOfElementLocated(passwordLocator));
            driver.findElement(passwordLocator).sendKeys("Password123");
            TimeUnit.SECONDS.sleep(3);

            wait.until(ExpectedConditions.elementToBeClickable(submitButtonLocator));
            driver.findElement(submitButtonLocator).click();
            TimeUnit.SECONDS.sleep(3);

            // Verify error message for invalid username
            wait.until(ExpectedConditions.visibilityOfElementLocated(errorMessageLocator));
            WebElement errorMessage = driver.findElement(errorMessageLocator);

            if (errorMessage.isDisplayed()) {
                System.out.println("Test Passed: Error message is displayed.");
                System.exit(0);
            } else {
                System.out.println("Test Failed: Error message is not displayed.");
                System.exit(1);
            }
        } catch (Exception e) {
            e.printStackTrace();
            System.out.println("Test Failed due to an Exception.");
            System.exit(1);
        } finally {
            driver.quit();
        }
    }
}
